### PR TITLE
✨ Motor Type Getter

### PR DESCRIPTION
**Pull request by liam-teale**
#### Summary:
add API that gets the type of a motor

#### Motivation:
V5 Motors and EXP Motors have slightly different behaviour. For example, EXP motors achieve their max velocity at 8v, while V5 motors achieve their max velocity at 12v. The user may wish to handle this difference, but this requires that they know the type of the motor that's currently plugged in

#### Test Plan:

- [x] motor_get_type returns MOTOR_TYPE_V5 if a v5 motor is plugged in
- [x] motor_get_type returns MOTOR_TYPE_EXP if an EXP motor is plugged in
- [x] motor_get_type returns MOTOR_TYPE_INVALID if no motor is plugged in
- [x] motor_get_type returns MOTOR_TYPE_INVALID if something other than a motor is plugged in
- [x] above tests succeed when using Motor::get_type
- [x] above tests succeed when using Motor::get_type_all
- [x] above tests succeed when using MotorGroup::get_type
- [x] above tests succeed when using MotorGroup::get_type_all

### CODE REVIEW

File: src/devices/vdml_motors.c [hunk 1]
```diff
@@ -310,3 +310,11 @@ int32_t motor_get_voltage_limit(int8_t port) {
 	int32_t rtn = vexDeviceMotorVoltageLimitGet(device->device_info);
 	return_port(port - 1, rtn);
 }
+
+motor_type_e_t motor_get_type(int8_t port) {
+	port = abs(port);
+	claim_port_i(port - 1, E_DEVICE_MOTOR);
+	//int32_t rtn = vexDeviceMotorTypeGet(device->device_info);
```
Comment (Rocky14683) [inline src/devices/vdml_motors.c]: is this by accident?

**Review by Rocky14683 — CHANGES_REQUESTED** [review]

File: src/devices/vdml_motors.c (hunk 1 above)
Comment (liam-teale) [reply src/devices/vdml_motors.c]: oh yea it is, one moment

**Comment by Rocky14683** [comment]
[Download link](https://artprodcus3.artifacts.visualstudio.com/A268c8aad-3bb0-47d2-9a57-cf06a843d2e8/a421d041-2579-4bf8-a0aa-d7a0307dade3/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL3B1cmR1ZS1hY20tc2lnYm90cy9wcm9qZWN0SWQvYTQyMWQwNDEtMjU3OS00YmY4LWEwYWEtZDdhMDMwN2RhZGUzL2J1aWxkSWQvNjU5My9hcnRpZmFjdE5hbWUvdGVtcGxhdGU1/content?format=zip)

**Comment by Rocky14683** [comment]
[download link](https://artprodcus3.artifacts.visualstudio.com/A268c8aad-3bb0-47d2-9a57-cf06a843d2e8/a421d041-2579-4bf8-a0aa-d7a0307dade3/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL3B1cmR1ZS1hY20tc2lnYm90cy9wcm9qZWN0SWQvYTQyMWQwNDEtMjU3OS00YmY4LWEwYWEtZDdhMDMwN2RhZGUzL2J1aWxkSWQvNjYyMS9hcnRpZmFjdE5hbWUvdGVtcGxhdGU1/content?format=zip)

**Review by Rocky14683 — APPROVED** [review]
LGTM for beta, for actual release we need to add alias for enums

**Pull request merged**